src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -617,9 +678,25 @@ internal void SearchAutoCompleteCandidates(string input)
                 }
                 else
                 {
-                    // Providing the saved search results to limit the scope of the query search.
-                    // Then add back the ML info on filterted nodes as the Search function accepts elements of type NodeSearchElement
-                    var foundNodes = Search(input, searchElementsCache.Select(x => x.Model));
+                    LuceneSearchUtilityNodeAutocomplete = new LuceneSearchUtility(dynamoViewModel.Model);
+
+                    //The dirName parameter doesn't matter because we are using RAMDirectory indexing and no files are created
+                    LuceneSearchUtilityNodeAutocomplete.InitializeLuceneConfig(string.Empty, LuceneSearchUtility.LuceneStorage.RAM);
+
+                    //Memory indexing process for Node Autocomplete (indexing just the nodes returned by the NodeAutocomplete service so we limit the scope of the query search)
+                    foreach (var node in searchElementsCache.Select(x => x.Model))
+                    {
+                        var doc = LuceneSearchUtility.InitializeIndexDocumentForNodes();
+                        AddNodeTypeToSearchIndex(node, doc);
+                    }
+
+                    //Write the Lucene documents to memory
+                    LuceneSearchUtilityNodeAutocomplete.CommitWriterChanges();
+
+                    var luceneResults = SearchNodeAutocomplete(input, true);
+                    var foundNodesModels = luceneResults.Select(x => x.Model).Distinct();
```

Comment:
How about here? Same that Distinct() may be skipped?

Reply:
This Dictint() was removed so we will be aware of when duplicated info was indexed.
commit: https://github.com/DynamoDS/Dynamo/pull/14169/commits/df86894df4df28bfabc6c2c77713e5a8cf2faeed